src/xPDO/xPDO.php

```diff
@@ -147,9 +148,9 @@ class xPDO {
      */
     private $cachePath= null;
     /**
-     * @var array An array of supplemental service classes for this xPDO instance.
+     * @var xPDOContainer A container of supplemental service classes for this xPDO instance.
      */
-    public $services= array ();
+    public $services= null;
```

Comment:
It shouldn't break any usages since it implements \ArrayAccess. But this then conflicts with it needing to be ContainerInterface, as you suggested.